### PR TITLE
Update SE for Symfony 3.0

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

**Comment by gnugat** [comment]
Test classes could be autoloaded only in a development environment by moving them from `src/AppBundle/Tests` to `tests` and adding a `autoload-dev` section in composer.
What do you think about this suggestion?

**Comment by stof** [comment]
I agree with @gnugat here. Separating tests from the runtime source is a good idea IMO

File: .gitignore [hunk 1]
```diff
@@ -1,11 +1,10 @@
-/web/bundles/
 /app/bootstrap.php.cache
-/app/cache/*
 /app/config/parameters.yml
-/app/logs/*
-!app/cache/.gitkeep
-!app/logs/.gitkeep
 /app/phpunit.xml
```
Comment (stof) [inline .gitignore]: should be changed too as we moved the phpunit config at the root

**Comment by stof** [comment]
this also requires an update of the HttpKernel component to change the location of the cache and logs

**Comment by weaverryan** [comment]
@gnugat @stof @fabpot is just making the structure match what was decided in #584 - so we should focus only on that (and open a new issue about tests for discussion).

**Comment by Tobion** [comment]
Created #855 for tests

**Comment by fabpot** [comment]
Let's move forward, merging now.

**Comment by fabpot** [comment]
Needs more work

File: .gitignore [hunk 2]
```diff
@@ -1,11 +1,10 @@
-/web/bundles/
 /app/bootstrap.php.cache
```
Comment (hacfi) [inline .gitignore]: this is moved to `/var/` as well

File: phpunit.xml.dist [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!-- http://phpunit.de/manual/4.1/en/appendixes.configuration.html -->
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.1/phpunit.xsd"
+         backupGlobals="false"
+         colors="true"
+         bootstrap="app/bootstrap.php.cache"
```
Comment (hacfi) [inline phpunit.xml.dist]: `bootstrap="var/bootstrap.php.cache"`

File: var/bootstrap.php.cache [hunk 4]
```diff
@@ -0,0 +1,3078 @@
+<?php
```
Comment (hacfi) [inline var/bootstrap.php.cache]: No need to commit this

**Comment by Tobion** [comment]
I'd recommend to merge #854 first.

File: bin/console [hunk 5]
```diff
@@ -7,8 +7,8 @@
 
 set_time_limit(0);
 
-require_once __DIR__.'/bootstrap.php.cache';
-require_once __DIR__.'/AppKernel.php';
+require_once dirname(__DIR__).'/app/bootstrap.php.cache';
```
Comment (weaverryan) [inline bin/console]: As others have said, this lives in `/var` - and the same change will need to go in the front controllers.

**Comment by fabpot** [comment]
Besides the new directory structure, I've made another change that should help: sessions are now stored outside of the `cache/` directory, and directly into the `var/` dir.

**Comment by Pierstoval** [comment]
@fabpot Why is it better to store the sessions in `var` instead of `cache` ?

**Comment by fabpot** [comment]
@Pierstoval Because clearing the session store should be decoupled from clearing the Symfony cache, the two are not related. And when using `cache:clear`, all the cache is flushed. Also, storing sessions in the cache has some problems with Windows (see symfony/symfony#15862 -- not confirmed though)

**Comment by fabpot** [comment]
This one is ready for a last review before merge.

File: bin/symfony_requirements [hunk 6]
```diff
@@ -1,6 +1,7 @@
+#!/usr/bin/env php
 <?php
```
Comment (yguedidi) [inline bin/symfony_requirements]: Are you sure about the file permissions change?

**Comment by michaelcullum** [comment]
Should `var/sessions/` not also have a `.gitkeep` and be added to the `.gitignore`?

**Comment by fabpot** [comment]
@michaelcullum right, fixed

**Pull request merged**